pxr/usd/sdf/path.h

```diff
@@ -806,6 +806,29 @@ class SdfPath : boost::totally_ordered<SdfPath>
     /// This returns \c true if IsValidIdentifier() does.
     SDF_API static bool IsValidNamespacedIdentifier(const std::string &name);
 
+    /// Returns whether \p name is a legal prim name for any path component.
+    SDF_API static bool IsValidPrimName(const std::string& name);
+
+    /// Returns whether \p name is a legal prim name for any path component.
+    /// This method uses the value of \p utf8 to determine which rules to
+    /// use to validate, regardless of the value of TF_UTF8_IDENTIFIERS.
+    SDF_API static bool IsValidPrimName(const std::string& name, bool utf8);
```

Comment:
I think I'd leave the `utf8` flag out of the Path API.  It seems like something that would just have to be deprecated once the UTF8 env setting is removed.